cmake/podioMacros.cmake

```diff
@@ -297,21 +193,10 @@ function(PODIO_ADD_ROOT_IO_DICT dict_name CORE_LIB HEADERS SELECTION_XML)
   # the core headers
   LIST(FILTER HEADERS EXCLUDE REGEX .*SIOBlock.h)
 
-  add_library(${dict_name} SHARED)
-  target_link_libraries(${dict_name} PUBLIC
-    ${CORE_LIB}
-    podio::podio
-    ROOT::Core
-    )
-  target_include_directories(${dict_name} PUBLIC
-    $<BUILD_INTERFACE:${ARG_OUTPUT_FOLDER}>
-    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+  ROOT_GENERATE_DICTIONARY(${dict_name} ${HEADERS} LINKDEF ${selectionfile}
```

Comment:
This also doesn't seem to create a cmake target(?) At least our EDM4hep tests fail because they can no longer find the target: https://github.com/AIDASoft/podio/actions/runs/5714684545/job/15482494614?pr=460#step:6:979

Reply:
It doesn't because it changes the way how the library itself is compiled (linking the dictionary inside for the moment)